src/main/java/org/broadinstitute/hellbender/utils/bigquery/CommittedBQWriter.java

```diff
@@ -57,45 +56,85 @@ protected void createStream() throws Descriptors.DescriptorValidationException,
+            // Google BigQuery write API error handling
+            // https://cloud.google.com/bigquery/docs/write-api#error_handling
+            // The comments in the case matching below are nearly all quotations from the documentation linked above.
+            Status.Code code = se.getStatus().getCode();
+            switch (code) {
+                case ALREADY_EXISTS:
+                    // ALREADY_EXISTS: The row was already written. This error can happen when you provide stream offsets.
+                    // It indicates that a duplicate record was detected. You can safely ignore this error.
+
+                    // We don't expect to see this, but to "ignore" we should not retry sending the same `jsonArr`.
+                    jsonArr = new JSONArray();
+                    return null;
+                case INVALID_ARGUMENT:
+                case NOT_FOUND:
+                case OUT_OF_RANGE:
+                case PERMISSION_DENIED:
+                    // Do not retry these:
+                    //
+                    // INVALID_ARGUMENT: Invalid argument. This error is an application error.
+                    //
+                    // NOT_FOUND: The stream or table was not found.
+                    //
+                    // OUT_OF_RANGE. The offset is beyond the current write offset. This error can happen if you provide
+                    // stream offsets and a previous write operation failed. In that case, you can retry from the last
+                    // successful write. This error can also happen if the application sets the wrong offset value.
+                    //
+                    // PERMISSION_DENIED. The application does not have permission to write to this table.
+                    throw e;
+                default:
+                    if (retryCount >= maxRetries) {
+                        throw new GATKException("Caught exception writing to BigQuery and " + maxRetries + " write retries are exhausted", e);
+                    }
+                    logger.warn("Caught exception writing to BigQuery, " + (maxRetries - retryCount - 1) + " retries remaining.", e);
+                    long backOffMillis = backoff.nextBackOffMillis();
+                    Thread.sleep(backOffMillis);
+
+                    switch (code) {
+                        case INTERNAL:
+                        case CANCELLED:
+                        case ABORTED:
+                            // INTERNAL, CANCELLED, or ABORTED: The operation could not be completed. You can safely retry the operation.
+                            // no extra work required here, break out to the recursion.
+                            break;
+                        default:
+                            // If you receive an error that's not listed above, then try to open a new connection by closing the
+                            // writer object and creating a new instance.
+                            writer.close();
```

Comment:
Not sure I understand the situation under which this might happen. Maybe put some logging here?

Reply:
There's already logging on line 115 for all cases, did you mean something additional for this particular scenario? This represents a significant fraction of the errors we see thanks mostly to `UNAVAILABLE` which tbh doesn't really mean anything different to me than the other codes. 😄